src/main/groovy/edu/oregonstate/mist/cardsapi/resources/CardsResource.groovy

```diff
@@ -65,4 +81,162 @@ class CardsResource extends Resource {
         }
 
     }
+
+    //Get cards by parameters
+    @GET
+    @Path ('')
+    @Produces(MediaType.APPLICATION_JSON)
+    Response getCards(@QueryParam("types") List<String> types,
+                      @QueryParam("name") Optional<String> name,
+                      @QueryParam("colors") List<String> colors,
+                      @QueryParam("rarities") List<String> rarities,
+                      @QueryParam("energyMin") Optional<Integer> energyMin,
+                      @QueryParam("energyMax") Optional<Integer> energyMax,
+                      @QueryParam("keywords") List<String> keywords,
+                      @QueryParam("number") Optional<Integer> number,
+                      @QueryParam("isRandom") Optional<Boolean> isRandom) {
+
+        Response response
+
+        List<String> validTypes = ["skill", "attack", "power", "status", "curse"]
+        List<String> validColors = ["red", "green", "blue", "colorless"]
+        List<String> validRarities = ["basic", "common", "uncommon", "rare"]
+
+        if(types.empty) {
+            types = validTypes
+        } else {
+            for(int i = 0; i < types.size(); i++) {
+                if(!validTypes.contains(types[i])) {
+                    response = badRequest("Invalid types").build()
+                    return response
+                }
+            }
+        }
+
+        if(colors.empty) {
+            colors = validColors
+        } else {
+            for(int i = 0; i < colors.size(); i++) {
+                if(!validColors.contains(colors[i])) {
+                    response = badRequest("Invalid colors").build()
+                    return response
+                }
+            }
+        }
+
+        if(rarities.empty) {
+            rarities = validRarities
+        } else {
+            for(int i = 0; i < rarities.size(); i++) {
+                if(!validRarities.contains(rarities[i])) {
+                    response = badRequest("Invalid rarities").build()
+                    return response
+                }
+            }
+        }
+
+        if(!(name.or("")).matches('[a-zA-Z0-9 ."+-]*')) {
+            response = badRequest("Invalid name").build()
```

Comment:
How about having a more specific message for why the name is invalid? Same comment for the keywords

Reply:
Also, you might want to make these regex patterns class-level variables since POST and PUT will probably need to use them too when creating/updating cards